posts/the-ring-of-integers-of-a-cyclotomic-field.md

```diff
@@ -0,0 +1,69 @@
+---
+author: 'Riccardo Brasca'
+category: 'New in mathlib'
+date: 2022-06-14 10:24:12 UTC+02:00
+description: ''
+has_math: true
+link: ''
+slug: the-ring-of-integers-of-a-cyclotomic-field
+tags: ''
+title: The ring of integers of a cyclotomic field
+type: text
+---
+In [PR #13585](https://github.com/leanprover-community/mathlib/pull/13585) we compute the
+discriminant of cyclotomic fields. This is an important result, usually treated in a first year
+graduate course in number theory. In this post we would like to explain why it is an important
+result, and briefly explain how we proved it.
+
+Number fields, finite extensions of `ℚ`, are fundamental objects in modern number theory.
+To give a simple application, let us consider the equation `x² + 2 = y³`, where we are
+looking for integer solutions. It's not easy to treat this equation directly in `ℤ`, since
+the left hand side is irreducible. On the other hand, working in `ℚ(√-2)` (the finite extension
+of `ℚ` obtained adding a root of `x² + 2`), the equation becomes `(x + √-2)(x - (√-2)) = y³`.
+We have now written a cube as a product and we can study the relations between the prime factors
+of the left and of the right hand side. One problem is that `ℚ(√-2)` is a field, so there is no
+useful notion of prime factors. The idea is that the same decomposition holds in `ℤ[√-2]` (the
+smallest subring of `ℂ` containing `√-2`).  This is a ring that "looks like `ℤ`" (to be precise
+it is a euclidean domain), where we can do arithmetic and solve the equation (whose only solution
+is `x = ±5` and `y = 3`).
+
+In the [flt-regular](https://github.com/leanprover-community/flt-regular) project we consider the equation
+`x ^ p + y ^ p = z ^ p` over `ℤ`, where `p` is a (regular, but this is irrelevant for our discussion)
+prime. The natural idea is to consider the field `ℚ(ζₚ)` and the ring `ℤ[ζₚ]`, where `ζₚ` is a primitive
```

Comment:
I think you don't give a name to `ℚ(ζₚ)` in this paragraph and later refers to cyclotomic fields.